### PR TITLE
Remove `start-node.sh` from Dockerfile

### PR DESCRIPTION
**Pull request by kkovaacs**
Now that `pathfinder` itself supports using environment variables instead of command-line arguments we can remove `start-node.sh` and just use the pathfinder binary directly in the image.

This PR also fixes a few small things:
  * An issue where even though we ran `tini`, it was _not_ the init process of the container rendering it pretty much useless.
  * Defaults the pathfinder JSON-RPC endpoint address so that it listens on `0.0.0.0` by default. This is required so that `EXPOSE` works out of the box.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -186,14 +186,21 @@ The `pathfinder` node can be run in the provided Docker image.
...
+  -e PATHFINDER_ETHEREUM_API_URL="https://goerli.infura.io/v3/<project-id>" \
+  eqlabs/pathfinder
 ```
 
+The following environment variables can be passed to the container:
+
+| Name                               | Description                                                  | Default value | Required |
+| ---------------------------------- | ------------------------------------------------------------ | ------------- | -------- |
+| PATHFINDER_ETHEREUM_API_URL        | Ethereum (archive) node JSON-RPC endpoint URL                |               | yes      |
+| PATHFINDER_ETHEREUM_API_USER_AGENT | User-Agent header value to use with Ethereum node API        |               | no       |
```
Comment (Mirko-von-Leipzig) [inline README.md:200]: I presume this means this can only be merged after #223 then?

**Review by Mirko-von-Leipzig — COMMENTED** [review]
I don't have the docker knowledge to really review this, but looks good to me :)

**Review by aphelionz — DISMISSED** [review]

**Review by koivunej — CHANGES_REQUESTED** [review]
PR needs to be recreated to target main, `start-node.sh` should be removed as well.

**Comment by kkovaacs** [comment]
Since this depended on PR #223 I created the PR basing it on that branch. GitHub automatically changes the target branch to `main` if that one you specified as the target of your PR gets merged.

**Comment by kkovaacs** [comment]
@koivunej I've removed `start-node.sh`, thanks for spotting this!

File: README.md (hunk 1 above)
Comment (kkovaacs) [reply README.md:200]: Yes.

**Review by CHr15F0x — APPROVED** [review]
LGTM, +1 for the ascii-art-table

**Review by koivunej — APPROVED** [review]

**Comment by kkovaacs** [comment]
@koivunej and if we consider how you start the Docker image as part of the CLI interface -- that's a breaking change, too!

@aphelionz Are you OK with the removal of IDLE?

**Comment by aphelionz** [comment]
Yes as long as its removal is matched (or remains matched if it was never added) in the README etc

**Comment by kkovaacs** [comment]
IDLE wasn't mentioned in the README.

**Pull request merged**